lib/utils.dart

```diff
@@ -131,7 +118,36 @@ PageLoaderElement rootElementOf(item) {
   try {
     return item.$root;
   } catch (_) {
-    throw "'rootElementOf' may only be called on PageObjects. "
-        "'${item.runtimeType}' is not a PageObject.";
+    throw new PageLoaderUtilsError.rootElementOfError();
   }
 }
+
+class PageLoaderUtilsError extends Error {
+  final String message;
+
+  PageLoaderUtilsError._(this.message);
+
+  static String _message(String f) => "'$f' may only be called on PageObjects"
+      'or PageLoaderElements';
+
+  factory PageLoaderUtilsError.existsError() =>
+      new PageLoaderUtilsError._(_message('exists/notExists'));
```

Comment:
All of the `PageLoaderUtilsError` errors are actually [ArgumentError](https://api.dartlang.org/dev/2.0.0-dev.55.0/dart-core/ArgumentError-class.html)s.

How about doing either of these?

(A) Just use [ArgumentError](https://api.dartlang.org/dev/2.0.0-dev.55.0/dart-core/ArgumentError-class.html). Make the factories return `ArgumentError` instances. That means that the class won't be useful other than to hold the factory functions so it could be declared `abstract`. Or you could do away with the class and just define a bunch of top-level factory functions.

&nbsp;&nbsp;Or alternatively:

(B) Renaming and rebasing the class (since "UtilsError" sounds like something broke in the utils lib):

- Renaming the class to `PageLoaderArgumentError`
- Extend `ArgumentError`
- Drop `final String message` and the `toString()` override


I think that (A) with the class declared abstract is a good option.

cc @kevmoo @kwalrath

Reply:
I'm more in favor of (B) since it keeps the error as its own type while also being a subtype of `ArgumentError`.